src/EventListener/LogRevisionsListener.php

```diff
@@ -383,8 +389,15 @@ private function getRevisionId(Connection $conn)
             );
 
             $revisionId = $conn->lastInsertId();
-            if (false === $revisionId) {
-                throw new \RuntimeException('Unable to retrieve the last revision id.');
+            /**
+             * Preceding lastInsertId throws Doctrine\DBAL\Exception\DriverException on doctrine/dbal 4+, making the next check unnecessary.
+             *
+             * NEXT_MAJOR: Remove the following block
+             */
+            if (\version_compare(InstalledVersions::getVersion('doctrine/dbal') ?? '', '4.0.0', '<')) {
```

Comment:
No need for this if.

The check will just be usefull for 3.x.

You can just write
```
if (false === $revisionId) { // @phpstan-ignore-line doctrine/dbal 3 lastInsertId() can return false
                    throw new \RuntimeException('Unable to retrieve the last revision id.');
                }
```